### PR TITLE
auxiliary function 'try_blas_flags' added.

### PR DESCRIPTION
**Pull request by mohammadpz**
Regarding issue #3744 and PR #3654 .
@nouiz , could you please take a look?

### CODE REVIEW

**Comment by mohammadpz** [comment]
In the line 2082 of ```theano/gof/cmodule```, in the function ```try_flags```, the argument ```try_run``` is False. While we need to ```try_run``` for blas. One way is to leave this as is while another way is to override the function ```try_flags```.

File: theano/tensor/blas.py [hunk 1]
```diff
@@ -344,6 +350,9 @@ def default_blas_ldflags():
     res = GCC_compiler.try_compile_tmp(
         test_code, tmp_prefix='try_blas_',
         flags=flags, try_run=True)
+    # res[0]: shows successful compilation
+    # res[1]: shows successful execution
+    print(res)
```
Comment (nouiz) [inline theano/tensor/blas.py]: remove the print.

File: theano/tensor/blas.py [hunk 2]
```diff
@@ -304,14 +305,16 @@ def default_blas_ldflags():
...
+                ret += ['-lblas']
                 return ' '.join(ret)
             # Try to add the anaconda lib directory to runtime loading of lib.
             # This fix some case with Anaconda 2.3 on Linux.
             if "Anaconda" in sys.version and "linux" in sys.platform:
                 lib_path = os.path.join(sys.prefix, 'lib')
                 ret.append('-Wl,-rpath,' + lib_path)
-                if GCC_compiler.try_flags(ret):
+                if GCC_compiler.try_flags(ret) and try_blas_flag():
+                    ret += ['-lblas']
```
Comment (nouiz) [inline theano/tensor/blas.py]: remove this addition to the flags. We already found the flags we think should be used, don't modify them. Idem elsewhere.

File: theano/tensor/blas.py [hunk 3]
```diff
@@ -304,14 +305,16 @@ def default_blas_ldflags():
                 ['-L%s' % l for l in blas_info['library_dirs']] +
                 ['-l%s' % l for l in blas_info['libraries']] +
                 [])
-            if GCC_compiler.try_flags(ret):
+            if GCC_compiler.try_flags(ret) and try_blas_flag():
```
Comment (nouiz) [inline theano/tensor/blas.py]: Doing those 2 tests are do duplicate test. Just call try_blas_flag, is cover what try_flags was doing.

But modify try_blas_flag() to take ret as a parameter and use it.

**Comment by nouiz** [comment]
thanks for the PR. I got 2 small comments.

File: theano/tensor/blas.py (hunk 3 above)
Comment (mohammadpz) [inline theano/tensor/blas.py]: @nouiz , For ```blas```, we also check the execution while for other flags we only check the compilation. Maybe we can also check execution for other flags too?

File: theano/tensor/blas.py [hunk 4]
```diff
@@ -322,7 +322,10 @@ def default_blas_ldflags():
     # readily available. We try to see if that's the case, rather
     # than disable blas. To test it correctly, we must load a program.
     # Otherwise, there could be problem in the LD_LIBRARY_PATH.
+    return try_blas_flag()
```
Comment (nouiz) [inline theano/tensor/blas.py]: this should be `return try_blas_flags(['-lblas'])

**Comment by nouiz** [comment]
Try it with the script I told you. Do you still get the same result?

**Comment by mohammadpz** [comment]
By running
```python
python theano/misc/check_blas.py -B 500
```
It leads to the same results.

**Pull request merged**